dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/McpServletInitializer.java

```diff
@@ -56,7 +56,8 @@ public void onStartup(Set<Class<?>> c, ServletContext context) throws ServletExc
                 Tr.info(tc, "MCP server endpoint: " + fullMcpUrl);
             }
         } catch (IllegalStateException e) {
-            Tr.warning(tc, "CWMCM0017W.cdi.inactive.warning"); // called if ToolRegistry.get() has an issue with CDI
+            String inactiveCdiMsg = "The MCP server endpoint for the application {0} is unavailable due to CDI being inactive. Verify that any MCP annotations are placed on methods of CDI beans that have an appropriate scope annotation (for example, @ApplicationScoped).";
+            Tr.event(tc, inactiveCdiMsg, context.getServletContextName()); // called if ToolRegistry.get() has an issue with CDI
```

Comment:
Trace messages don't do placeholder substitutions. The data objects get printed on the next lines.

Since it's a trace message, you can be more brief here.

```suggestion
            Tr.event(this, tc, "CDI not active, not registering MCP endpoint", context.getServletContextName()); // called if ToolRegistry.get() has an issue with CDI
```

Usually, [trace logs should have an `if` block around them](https://github.com/OpenLiberty/open-liberty/wiki/Logging-and-Trace#adding-trace-to-your-classes) for performance reasons. That's probably not a concern, but you should be aware in case you're adding trace anywhere else.

There are [templates](https://github.com/OpenLiberty/open-liberty/wiki/Logging-and-Trace#eclipse-templates) available to add the trace line with the `if` block.